packages/rpc-methods/src/restricted/dialog.test.ts

```diff
@@ -0,0 +1,309 @@
+        undefined,
+        null,
+        false,
+        '',
+        2,
+        [],
+        {},
+        new (class {})(),
+        new Array(41).fill('a').join(''),
+      ]) {
+        await expect(
+          implementation({
+            context: { origin: 'foo' },
+            method: 'snap_dialog',
+            params: {
+              type: DialogType.alert,
+              fields: {
+                title: invalidInput,
+                description: 'Bar',
+                textAreaContent: 'Baz',
+              } as any,
+            },
+          }),
+        ).rejects.toThrow(
+          'Must specify a non-empty string "title" less than 40 characters long.',
+        );
+      }
+    });
+
+    it('rejects invalid descriptions', async () => {
+      const hooks = getMockDialogHooks();
+      const implementation = getDialogImplementation(hooks);
+
+      for (const invalidInput of [
+        true,
+        2,
+        [],
+        {},
+        new (class {})(),
+        new Array(141).fill('a').join(''),
+      ]) {
+        await expect(
+          implementation({
+            context: { origin: 'foo' },
+            method: 'snap_dialog',
+            params: {
+              type: DialogType.alert,
+              fields: {
+                title: 'Foo',
+                description: invalidInput,
+                textAreaContent: 'Baz',
+              } as any,
+            },
+          }),
+        ).rejects.toThrow(
+          '"description" must be a string no more than 140 characters long if specified.',
+        );
+      }
+    });
+
+    it('rejects invalid text area contents', async () => {
+      const hooks = getMockDialogHooks();
+      const implementation = getDialogImplementation(hooks);
+
+      for (const invalidInput of [
+        true,
+        2,
+        [],
+        {},
+        new (class {})(),
+        new Array(1801).fill('a').join(''),
+      ]) {
+        await expect(
+          implementation({
+            context: { origin: 'foo' },
+            method: 'snap_dialog',
+            params: {
+              type: DialogType.alert,
+              fields: {
+                title: 'Foo',
+                description: 'Bar',
+                textAreaContent: invalidInput,
+              } as any,
+            },
+          }),
+        ).rejects.toThrow(
+          '"textAreaContent" must be a string no more than 1800 characters long if specified.',
+        );
+      }
+    });
+
+    it('rejects textAreaContent field for prompts', async () => {
+      const hooks = getMockDialogHooks();
+      const implementation = getDialogImplementation(hooks);
+      await expect(
+        implementation({
+          context: { origin: 'foo' },
+          method: 'snap_dialog',
+          params: {
+            type: DialogType.prompt,
+            fields: {
+              title: 'Foo',
+              description: 'Bar',
+              textAreaContent: 'Baz',
+            } as any,
```

Comment:
I think we should prefer `// @ts-expect-error` over casting to `any`. The benefit is that it would catch situations where we change something, so the input is suddenly accepted without a type cast. `@ts-expect-error` results in an error if the code _does not_ cause errors.

Reply:
You need to fight @ritave about this, because he has been asking me to cast to `any` instead of using the flag.

Reply:
It's a little bit annoying to do `// @ts-expect-error` here because there are two errors, and I think `any` is fine in tests. That said, I think we should prefer the comment in application code unless we have a specific reason not to use it.